static/app/stores/teamStore.tsx

```diff
@@ -37,24 +41,47 @@ const teamStoreConfig: Reflux.StoreDefinition & TeamStoreInterface = {
     this.listenTo(TeamActions.createTeamSuccess, this.onCreateSuccess);
     this.listenTo(TeamActions.fetchDetailsSuccess, this.onUpdateSuccess);
     this.listenTo(TeamActions.loadTeams, this.loadInitialData);
+    this.listenTo(TeamActions.loadUserTeams, this.loadUserTeams);
     this.listenTo(TeamActions.removeTeamSuccess, this.onRemoveSuccess);
     this.listenTo(TeamActions.updateSuccess, this.onUpdateSuccess);
   },
 
   reset() {
-    this.state = {teams: [], loading: true, hasMore: null};
+    this.state = {teams: [], loadedUserTeams: false, loading: true, hasMore: null};
   },
 
   loadInitialData(items, hasMore = null) {
     this.initialized = true;
     this.state = {
       teams: items.sort((a, b) => a.slug.localeCompare(b.slug)),
+      // TODO(davidenwang): Replace with a more reliable way of knowing when we have loaded all teams
+      loadedUserTeams: items.length < MAX_TEAMS,
```

Comment:
loadInitialData is called on app startup (from `loadTeamsAndProjects`) and will fetch up to 100 teams. If we receive less than 100 teams it means that we have not hit that 100 team pagination limit and therefore have all the organization's teams which means we have the user's teams in there somewhere as well.